plugins/tech-insights/src/api/TechInsightsClient.ts

```diff
@@ -47,7 +50,14 @@ export class TechInsightsClient implements TechInsightsApi {
 
   async getAllChecks(): Promise<Check[]> {
     const url = await this.discoveryApi.getBaseUrl('tech-insights');
-    const response = await fetch(`${url}/checks`);
+    const token = await this.identityApi.getIdToken();
```

Comment:
`getIdToken` is deprecated in favor of `getCredentials`. It's all soon to be replaced by [`fetchApiRef`](https://github.com/backstage/backstage/pull/6399) though :grin:

Figure it's fine to merge as is as there's plenty of `getIdToken` usage that we'll need to take care of at once